### PR TITLE
Add flags to Jaeger helm chart to exclude Jaeger and opencensus collector

### PR DESCRIPTION
**Pull request by tarvip**
Improve linkerd-jaeger helm chart with additional flags.

Signed-off-by: Tarvi Pillessaar <tarvip@gmail.com>

### CODE REVIEW

**Comment by Pothulapati** [comment]
Should we have a new `jaeger.enabled` flag instead based on which jaeger install can be toggled? @alpeb 

We currently follow this pattern with `promethus`, `grafana` 🤔

**Comment by tarvip** [comment]
> Should we have a new `jaeger.enabled` flag instead based on which jaeger install can be toggled? @alpeb
> 
> We currently follow this pattern with `promethus`, `grafana` 🤔

I was also thinking about that approach.

**Comment by tarvip** [comment]
Maybe there should be also `collector.enabled` flag for cases when using also existing collector.

**Comment by alpeb** [comment]
Agreed, `collector.enabled` and `jaeger.enabled` are more consistent with viz' approach :+1:

**Comment by tarvip** [comment]
Updated pull request.

File: jaeger/charts/linkerd-jaeger/values.yaml [hunk 1]
```diff
@@ -15,6 +15,8 @@ nodeSelector: &default_node_selector
 tolerations: &default_tolerations
 
 collector:
+  # -- Set to false do exclude collector installation
```
Comment (alpeb) [inline jaeger/charts/linkerd-jaeger/values.yaml:18]:
```suggestion
  # -- Set to false to exclude collector installation
```

File: jaeger/charts/linkerd-jaeger/values.yaml [hunk 2]
```diff
@@ -32,6 +34,8 @@ collector:
   jaegerAddr:
 
 jaeger:
+  # -- Set to false do exclude all-in-one Jaeger installation
```
Comment (alpeb) [inline jaeger/charts/linkerd-jaeger/values.yaml:37]:
```suggestion
  # -- Set to false to exclude all-in-one Jaeger installation
```

**Review by alpeb — APPROVED** [review]
Looks good to me, modulo typos :wink: :+1:

**Review by kleimkuhler — APPROVED** [review]
Looks good @tarvip thanks! You may need to merge with the latest `main` to pick up some fixes in CI if it continues to fail.

**Comment by Pothulapati** [comment]
Hey @tarvip, Thanks for working on this.

Changes like these are a bit easy to get wrong i.e with Helm indentation. Unit tests really help in catching them. I've raised https://github.com/linkerd/linkerd2/pull/6166 to add install unit tests. Can you also add extra cases (once the PR is merged) there with jaeger and oc disabled independently, and add golden files for the same? Just so that it is clear on what the output manifests are?

Once again, Thanks for taking this up!

**Review by adleong — APPROVED** [review]

**Pull request merged**